amethyst_ui/src/text.rs

```diff
@@ -372,17 +373,18 @@ impl<'a> System<'a> for UiKeyboardSystem {
                                 })
                                 .collect::<Vec<_>>();
                             // In instances of ambiguity we want to select the element with the
-                            // lowest Z order, so we need to find the lowest Z order value among
-                            // eligible elements
-                            let lowest_z = eligible
+                            // highest Z order, so we need to find the highest Z order value among
+                            // eligible elements.
+                            let highest_z = eligible
                                 .iter()
-                                .fold(INFINITY, |lowest, &(_, t)| lowest.min(t.global_z));
+                                .fold(-INFINITY, |highest, &(_, t)| highest.max(t.global_z));
```

Comment:
can you use [`NEG_INFINITY`](https://doc.rust-lang.org/std/f32/constant.NEG_INFINITY.html) instead?

Reply:
done